tests/pipeline/test_engine.py

```diff
@@ -271,6 +315,123 @@ def test_numeric_factor(self):
             DataFrame(expected_avg, index=dates, columns=self.assets),
         )
 
+    def test_rolling_and_nonrolling(self):
+        open_ = USEquityPricing.open
+        close = USEquityPricing.close
+        volume = USEquityPricing.volume
+
+        # Test for thirty days up to the second to last day that we think all
+        # the assets existed.  If we test the last day of our calendar, no
+        # assets will be in our output, because their end dates are all
+        dates_to_test = self.dates[-32:-2]
```

Comment:
Assets should still appear on the last day of their lifetime I think?  They don't appear on the **first** day of their lifetime, because we don't have any prior trading days.